plugins/main/bridge/bridge.go

```diff
@@ -465,8 +461,35 @@ func cmdAdd(args *skel.CmdArgs) error {
 			if err := ipam.ConfigureIface(args.IfName, result); err != nil {
 				return err
 			}
+			return nil
+		}); err != nil {
+			return err
+		}
+
+		// check bridge port state
+		for i := 0; i < 3; i++ {
```

Comment:
It would be nice if this had some backoff - sleeping 50, then, 500, then 1000 ms.

A more idiomatic way of writing this might be
```go
retries := []int{50, 500, 1000, 1000}
for idx, sleep := range retries {
  (check host)
  if idx == len(retries) - 1) {
    fail
  }
  time.Sleep(sleep + time.Millisecond)
}
```

Reply:
@squeed    have been modified.